linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/JSGen.scala

```diff
@@ -385,7 +385,80 @@ private[emitter] final class JSGen(val semantics: Semantics,
   def genClassDataOf(className: ClassName)(implicit pos: Position): Tree =
     genClassDataOf(ClassRef(className))
 
-  def envModuleField(module: String)(implicit pos: Position): VarRef = {
+  // Compiler generated identifiers --------------------------------------------
+
+  /* We distinguish three types of compiler generated identifiers:
+   *
+   * - envVar: Things pertaining to a given class. Accessible from every chunk
+   *   of the Scala.js output by importing its module.
+   * - coreJSLibVar: Things defined by the CoreJSLib. Accessible from every
+   *   chunk by importing the root module.
+   * - fileLevelVar: Things that are local to an individual file.
+   */
+
+  def envVar(field: String, className: ClassName)(implicit pos: Position): VarRef =
+    fileLevelVar(field, genName(className))
```

Comment:
It's confusing that `envVar` and `coreJSLibVar` delegate to `fileLevelVar`, although those three appear to be on par with each other. Consider having all three delegate to a "generic" implementation, that is not directly associated with one of them.